
BA interview process: Home test


NAME: JOSE ARTURO VALDIVIA RIVERA
CI: 8350080

Part 1: Analysis of the state of the business.

1. Factors Affecting Profitability and Key Analysis Areas:
The profitability of the startup faces challenges from several key factors:

•	Rising Delinquency Rates (DQ): The increasing DQ rates for both Contigo A (13.5%) and Contigo B (8%) directly translate to higher losses. These rates mean a larger proportion of loans won't be repaid in full, cutting into revenue.

•	Ambitious Growth Plans: While growth is positive, the aggressive expansion, especially for Contigo A (25% monthly), can strain resources and potentially lead to a higher influx of riskier borrowers, further exacerbating the DQ issue, Increasing credit risk.

•	Competitive Market for Contigo B: The slower growth of Contigo B (10%
monthly) due to market competition could limit revenue potential from this less risky loan segment.

•	Operational Costs: Although scalable, the 30 pesos per account operational cost can become substantial as the customer base grows rapidly, especially with a high proportion of Contigo A customers.

**To analyze net income in the coming months, we should focus on:**

•	Revenue: Interest income from repaid loans, considering different interest rates and growth rates for Contigo A and B.

•	Losses: Amount of loan principal lost due to delinquencies, factored by DQ rates and loan values.

•	Operational Costs: Total cost calculated by multiplying the per-account cost by the total number of customers each month.

•	Net Income: Revenue minus losses and operational costs.

•	Market and Regulation framework: We should be ready for any change in the regulation and trying to adapt quickly in order to maintain revenue levels and control credit risk potencial



2. Is this startup going to be profitable during the next 12 months? What are the major trends driving
changes in net income? Suggestion: Provide a model for net income and forecast it for the next
12 months. Hint: losses come from the total debt not being repaid, and revenue comes from
collected interest. We will only consider operation costs as expenses.

In [6]:

import pandas as pd

# Initial customer data
data = {
    'Month': [1],
    'Contigo_A_Customers': [5000],
    'Contigo_B_Customers': [10000]
}

# Create DataFrame
df = pd.DataFrame(data)

# Growth rates
growth_A = 0.25
growth_B = 0.10

# Forecast customer base
for month in range(2, 13):
    df.loc[month - 1] = [
        month,
        int(df.iloc[month - 2]['Contigo_A_Customers'] * (1 + growth_A)),  # Compound growth
        int(df.iloc[month - 2]['Contigo_B_Customers'] * (1 + growth_B))    # Compound growth
    ]

# Loan amounts and interest rates
loan_A = 1000
loan_B = 2000
interest_A = 0.15
interest_B = 0.10

# Delinquency rates
dq_A = 0.135
dq_B = 0.08

# Operating cost per account
cost_per_account = 30

# Calculate financials
df['Revenue_A'] = (df['Contigo_A_Customers'] ) * loan_A * interest_A
df['Revenue_B'] = (df['Contigo_B_Customers'] ) * loan_B * interest_B
df['Losses_A'] = df['Contigo_A_Customers'] * dq_A * loan_A
df['Losses_B'] = df['Contigo_B_Customers'] * dq_B * loan_B
df['Expenses'] = (df['Contigo_A_Customers'] + df['Contigo_B_Customers']) * cost_per_account
df['Net_Income'] = df['Revenue_A'] + df['Revenue_B'] - df['Losses_A'] - df['Losses_B'] - df['Expenses']

print(df)

    Month  Contigo_A_Customers  Contigo_B_Customers  Revenue_A  Revenue_B  \
0       1                 5000                10000   750000.0  2000000.0   
1       2                 6250                11000   937500.0  2200000.0   
2       3                 7812                12100  1171800.0  2420000.0   
3       4                 9765                13310  1464750.0  2662000.0   
4       5                12206                14641  1830900.0  2928200.0   
5       6                15257                16105  2288550.0  3221000.0   
6       7                19071                17715  2860650.0  3543000.0   
7       8                23838                19486  3575700.0  3897200.0   
8       9                29797                21434  4469550.0  4286800.0   
9      10                37246                23577  5586900.0  4715400.0   
10     11                46557                25934  6983550.0  5186800.0   
11     12                58196                28527  8729400.0  5705400.0   

3. What changes are required to sustain the business not only throughout the year but in the long
run? Build a proposal of changes to main startup KPIs to achieve this goal. Discuss what
difficulties the startup may find in applying your recommendation.

Customer Segmentation and Risk Assessment Refinement: Develop more sophisticated models to better identify high-risk customers. This could involve using additional data points beyond just past delinquency.

Dynamic Pricing: Adjust interest rates based on individual customer risk profiles. This could help offset losses from higher-risk borrowers.

Operational Efficiency Improvements: Look for ways to reduce the operating cost per account.
Product Diversification: Consider offering different loan products (e.g., longer-term loans) or additional financial services to diversify revenue streams and reduce reliance on a single product.

Macroeconomics and Economy sectors: We could look to implement a new product with a larger loan amount focused on sectors with good performance: in my current bank we have special conditions for people working in the agricultural sector. We will need more information about the different sectors of the economy to have a model based on GDP and inflation, for example.

Marketing and Retention Strategies: Invest in targeted marketing to attract lower-risk customers and implement retention programs to keep valuable customers.
Difficulties in Implementation:

Data Limitations: Implementing sophisticated risk models and dynamic pricing might require more data than the startup currently has.
Competitive Pressure: Adjusting interest rates or loan terms might not be feasible in a highly competitive market.
Operational Challenges: Reducing operating costs can be difficult without sacrificing service quality or customer acquisition.

Invest: We should have a budget to invest to reduce the operative cost to have an account.

Part 2: Analysis of a collection strategy.

In [13]:
import pandas as pd

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

# Read the CSV file into a DataFrame
df_contactability_scores = pd.read_csv('/content/contactability_scores_(4)_(2).csv')

# Display the first 5 rows
print(df_contactability_scores.head().to_markdown(index=False, numalign="left", stralign="left"))

# Print the column names and their data types
print(df_contactability_scores.info())

| customer_id   | contactability_score   | total_debt   | interest   |
|:--------------|:-----------------------|:-------------|:-----------|
| 1             | 0.293                  | 1000         | 150        |
| 2             | 0.25                   | 1000         | 150        |
| 3             | 0.033                  | 1000         | 150        |
| 4             | 0.239                  | 1000         | 150        |
| 5             | 0.681                  | 1000         | 150        |
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 675 entries, 0 to 674
Data columns (total 4 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   customer_id           675 non-null    int64  
 1   contactability_score  675 non-null    float64
 2   total_debt            675 non-null    int64  
 3   interest              675 non-null    int64  
dtypes: float64(1), int64(3)
memory usage: 21.2 KB
None


Calculate the following KPIs:

a. Recovered accounts: The number of delinquent accounts successfully recovered. This is calculated by multiplying the total number of delinquent customers by the probability of contacting them and the probability of them paying once contacted.

b. Recovered total debt: The total amount of debt recovered from delinquent customers. This is calculated by multiplying the recovered accounts by the average total debt per customer.

c. Revenue from collected interest: The interest earned from the recovered debt. This is calculated by multiplying the recovered accounts by the average interest per customer.

d. Total campaign cost: The total cost of the campaign, which is the cost per call multiplied by the total number of delinquent customers.

e. DQ losses from uncollected total debt: The total amount of debt that remains uncollected after the campaign. This is calculated by subtracting the recovered total debt from the total initial delinquent debt.

f. Collections income: The net income from the collection campaign, calculated by subtracting the total campaign cost and the remaining DQ losses from the revenue from collected interest.

1. The probable results of the campaign, assuming it's launched to all 675 delinquent customers, are as follows:

In [14]:
# Calculate expected recovery
df_contactability_scores['expected_recovery'] =  df_contactability_scores['contactability_score'] * 0.75

# Calculate recovered accounts by summing up all expected recoveries
recovered_accounts = df_contactability_scores['expected_recovery'].sum()

# Calculate recovered debt
recovered_debt = (df_contactability_scores['expected_recovery'] * df_contactability_scores['total_debt']).sum()

# Calculate recovered interest
recovered_interest = (df_contactability_scores['expected_recovery'] * df_contactability_scores['interest']).sum()

# Calculate total campaign cost
total_campaign_cost = 675 * 30

# Calculate uncollected debt
uncollected_debt = df_contactability_scores['total_debt'].sum() - recovered_debt

# Calculate DQ losses
dq_losses = uncollected_debt * 0.1

# Calculate collections income
collections_income = recovered_interest - total_campaign_cost - dq_losses

# Print the KPIs
print("\n KPIs of the campaign")
print("--------------------------")
print(f'Recovered accounts: {recovered_accounts:.0f}')
print(f'Recovered debt: ${recovered_debt:.2f}')
print(f'Recovered interest: ${recovered_interest:.2f}')
print(f'Total campaign cost: ${total_campaign_cost:.2f}')
print(f'DQ losses: ${dq_losses:.2f}')
print(f'Collections income: ${collections_income:.2f}')


 KPIs of the campaign
--------------------------
Recovered accounts: 212
Recovered debt: $212024.25
Recovered interest: $31803.64
Total campaign cost: $20250.00
DQ losses: $46297.58
Collections income: $-34743.94


2. What is the impact delivered on overall business performance by this campaign? To answer this
question, we provide the results of month 1 without collections in the table below. Using the
results from the previous question and the data coming from the table, provide a justification for
applying or not this collection strategy.

In [16]:
# Create a dictionary with baseline performance
baseline_performance = {'Month': 1, 'Accounts': 5000, 'DQ accounts': 675,
                        'Interest revenue': 648750.00, 'DQ losses': 675000.00,
                        'Operative cost': 150000, 'Net income': -176250.00}

# Create a list with campaign performance
campaign_performance = [1, 5000, 675, recovered_interest, dq_losses,
                        total_campaign_cost, collections_income]

# Create a dataframe for comparison
df_comparison = pd.DataFrame(columns=baseline_performance.keys())
df_comparison.loc['Without collections'] = baseline_performance.values()
df_comparison.loc['With collections'] = campaign_performance

# Print the comparison dataframe
print(df_comparison.to_markdown(index=True, numalign="left", stralign="left", floatfmt='.2f'))

|                     | Month   | Accounts   | DQ accounts   | Interest revenue   | DQ losses   | Operative cost   | Net income   |
|:--------------------|:--------|:-----------|:--------------|:-------------------|:------------|:-----------------|:-------------|
| Without collections | 1.00    | 5000.00    | 675.00        | 648750.00          | 675000.00   | 150000.00        | -176250.00   |
| With collections    | 1.00    | 5000.00    | 675.00        | 31803.64           | 46297.58    | 20250.00         | -34743.94    |


From this table, we can observe the following:

DQ losses decrease significantly from $675,000 to $46,297.58, indicating a substantial reduction in losses due to delinquency

Net income improves from -$176,250 to -$34,743.94, although it remains negative
the first month
However, there are also downsides:

Interest revenue drops considerably from $648,750 to $31,803.64, likely because fewer accounts are in delinquency and accruing interest
Overall, the campaign helps in reducing DQ losses, which is the primary goal given the high DQ rate of Contigo A customers. However, it comes at the cost of reduced interest revenue. The decision to implement this strategy depends on whether the reduction in DQ losses outweighs the decrease in interest revenue.

If we do this example for a year or maybe many years its better to have a campain to reduce delinquency rates in the long run.

3. Discuss some possible improvements to this campaign.

Several strategies could be explored to improve the campaign's effectiveness and potentially make it profitable:

Targeted Calling: Instead of calling all delinquent customers, focus on those with a higher contactability score and/or a higher outstanding debt. This would increase the chances of recovering more debt while reducing the overall campaign cost.

Cost Reduction: Negotiate lower costs per call or explore alternative communication channels like SMS or email, which might be less expensive.
Incentives: Offer discounts or other incentives to encourage early repayment, which could increase the recovery rate and reduce losses.

Machine Learning Models: Develop more sophisticated models to predict not only contactability but also the likelihood of repayment after contact. This could help further refine the targeting strategy.

Customer Segmentation: Analyze the characteristics of delinquent customers to identify different segments and tailor collection strategies accordingly.

By implementing some of these improvements, the collection campaign could become a valuable tool for reducing delinquency rates and improving the overall profitability of the Contigo A product.